scripts/deterministic/DeployScroll.s.sol

```diff
@@ -0,0 +1,1272 @@
+        deployL1WETHGateway();
+        deployL1CustomERC20Gateway();
+        deployL1ERC721Gateway();
+        deployL1ERC1155Gateway();
+    }
+
+    // @notice deployL2Contracts2ndPass deploys L2 contracts whose initialization depends on some L1 addresses.
+    function deployL2Contracts2ndPass() private broadcast(Layer.L2) {
+        // upgradable
+        deployL2ScrollMessenger();
+        deployL2GatewayRouter();
+        deployL2StandardERC20Gateway();
+        deployL2ETHGateway();
+        deployL2WETHGateway();
+        deployL2CustomERC20Gateway();
+        deployL2ERC721Gateway();
+        deployL2ERC1155Gateway();
+    }
+
+    // @notice initializeL1Contracts initializes contracts deployed on L1.
+    function initializeL1Contracts() private broadcast(Layer.L1) only(Layer.L1) {
+        initializeScrollChain();
+        initializeL2GasPriceOracle();
+        initializeL1MessageQueue();
+        initializeL1ScrollMessenger();
+        initializeEnforcedTxGateway();
+        initializeL1GatewayRouter();
+        initializeL1CustomERC20Gateway();
+        initializeL1ERC1155Gateway();
+        initializeL1ERC721Gateway();
+        initializeL1ETHGateway();
+        initializeL1StandardERC20Gateway();
+        initializeL1WETHGateway();
+        initializeL1Whitelist();
+
+        transferL1ContractOwnership();
+    }
+
+    // @notice initializeL2Contracts initializes contracts deployed on L2.
+    function initializeL2Contracts() private broadcast(Layer.L2) only(Layer.L2) {
+        initializeL2MessageQueue();
+        initializeL2TxFeeVault();
+        initializeL1GasPriceOracle();
+        initializeL2ScrollMessenger();
+        initializeL2GatewayRouter();
+        initializeL2CustomERC20Gateway();
+        initializeL2ERC1155Gateway();
+        initializeL2ERC721Gateway();
+        initializeL2ETHGateway();
+        initializeL2StandardERC20Gateway();
+        initializeL2WETHGateway();
+        initializeScrollStandardERC20Factory();
+        initializeL2Whitelist();
+
+        transferL2ContractOwnership();
+    }
+
+    /***************************
+     * L1: 1st pass deployment *
+     ***************************/
+
+    function deployL1Weth() private {
+        L1_WETH_ADDR = deploy("L1_WETH", type(WrappedEther).creationCode);
+    }
+
+    function deployL1ProxyAdmin() private {
+        bytes memory args = abi.encode(DEPLOYER_ADDR);
+        L1_PROXY_ADMIN_ADDR = deploy("L1_PROXY_ADMIN", type(ProxyAdminSetOwner).creationCode, args);
+    }
+
+    function deployL1PlaceHolder() private {
+        L1_PROXY_IMPLEMENTATION_PLACEHOLDER_ADDR = deploy(
+            "L1_PROXY_IMPLEMENTATION_PLACEHOLDER",
+            type(EmptyContract).creationCode
+        );
+    }
+
+    function deployL1Whitelist() private {
+        bytes memory args = abi.encode(DEPLOYER_ADDR);
+        L1_WHITELIST_ADDR = deploy("L1_WHITELIST", type(Whitelist).creationCode, args);
+    }
+
+    function deployL2GasPriceOracle() private {
```

Comment:
it is not used if we deploy `L1MessageQueueWithGasPriceOracle`.

Reply:
Raised a pr here to resolve this: https://github.com/scroll-tech/scroll-contracts/pull/38